Process and Layer imagery/elevation

gchoqueux opened the pull request:
## Description
Work on `process` and `Layer` imagery/elevation:
* Move (see delete) parts of the code into more appropriate classes or clearer functions.
* Extract common methods.
* Fix

gchoqueux committed:
fix(process): pass, through the command, the parsedData attached to textures

gchoqueux committed:
fix(examples): dilate extent to Region WMS to show entire data.

the problem comes from LayeredMaterialNodeProcessing.updateLayeredMaterialNodeImagery.
No texture is provided for a tile, if only one of them is outside the source extent.

Visit PR #1067 https://github.com/iTowns/itowns/issues/1067 to follow the issue.

gchoqueux committed:
refactor(process): add default time Stamp parameter to LayerUpdateState.canTryUpdate.

gchoqueux committed:
refactor(layer): use layer.projection to add MaterialLayer and move colorLayersOrder in tileLayer

gchoqueux committed:
refactor(core): add zoom to each extent of tile

to follow issue visiting: https://github.com/iTowns/itowns/issues/1069

zarov commented on src/Process/LayeredMaterialNodeProcessing.js:
```diff
@@ -54,6 +43,36 @@ function refinementCommandCancellationFn(cmd) {
     return !cmd.requester.material.visible;
 }
 
+function handlingError(err, node, layer, targetLevel, view) {
```
Couldn't this be moved to `TileMesh` or in a `ProcessHelper` file along with `buildCommand` below ?

gchoqueux replied:
I move `handlingError` but not `buildCommand` is specify for each process

zarov replied:
Thanks

zarov commented on src/Converter/convertToTile.js:
```diff
@@ -45,12 +72,13 @@ export default {
-        tile.layers.set(layer.threejsLayer);
+
+        // TODO unecessary because is undefined...????
+        // tile.layers.set(layer.threejsLayer);
```
`threejsLayer` is used in `Picking` and for debug isn't it ? Do you have clue on how to maybe get rid of this property ?

gchoqueux replied:
It's used with visibility. This part needs to be fixed, because there's conflict with `object3d` added in `view.scene`;

zarov reviewed (commented):
Thanks for this first cleaning !

gchoqueux committed:
refactor(process): extract common functions in layered material processing.

extract common functions in layered material processing and feature processing:
   * The target is principally to highlight the common parties and simplify the refactoring in progress.

other:
   * remove useless test command parameter.

gchoqueux committed:
refactor(convert): in convert to tile mesh separate init tile from layer

zarov reviewed (approved):
LGTM
